src/interfaces/chain.cpp

```diff
@@ -248,6 +249,20 @@ class ChainImpl : public Chain
         // LockImpl to Lock pointer
         return std::move(result);
     }
+    Optional<bool> filterMatchesAny(const uint256& hash, const GCSFilter::ElementSet& filter_set) override
+    {
+        const BlockFilterIndex* block_filter_index = GetBlockFilterIndex(BlockFilterType::BASIC);
+        if (!block_filter_index) return {};
+        BlockFilter filter;
+        const CBlockIndex* index;
```

Comment:
This could lock even smaller block of code:
```
        {
            LOCK(cs_main);
            index = LookupBlockIndex(hash);
        }
        if (!index) return {};
```

Or maybe you would care to use `ES.28: Use lambdas for complex initialization, especially of const variables` guideline (note `const` for `index` pointer itself):
```
const CBlockIndex* const index = [&]{ 
    LOCK(cs_main);
    return LookupBlockIndex(hash);
}();
if (!index) return {};
```
```